1.0.5

ankgne opened the pull request:
Hi Juan,

I have done following enhancement in v1.0.5 

1. Added the functionality of selecting multiple comma separated terms 
2. Added the functionality of multiple comma separated tax 
3. Added the functionality of pagination and a new argument named 'pagination'. By default pagination is set to false. Pagination can be enabled by passing argument pagination=yes
4. Enhanced the logic of querying post . Existing post query does not check if there are any featured images attached to it. Enhanced the logic to return posts that have featured images attached to them. Enhanced logic fixes the issue of returning posts that do not have featured images and we display count of posts as selected by user in short code

Refactoring:

1. Created function for displaying filters
2. Created function for extracting argument of shortcode
3. Added private variables

ankgne committed:
1.0.5

Added the CSS for pagination functionality

ankgne committed:
1.0.5

ankgne committed:
1.0.5

Added the functionality of selecting multiple comma separated terms
Added the functionality of multiple comma separated tax
Added the functionality of pagination and a new argument named
'pagination'. By default pagination is set to false. Pagination can be
enabled by passing argument pagination=yes
Enhanced the logic of querying post . Existing post query does not check
if there are any featured images attached to it. Enhanced the logic to
return posts that have featured images attached to them. Enhanced logic
fixes the issue of returning posts that do not have featured images and
we display count of posts as selected by user in short code

Refactoring:
Created function for displaying filters
Created function for extracting argument of shortcode
Added private variables

ankgne committed:
1.0.5

Added new contributor and change log details of 1.0.5

ankgne committed:
1.0.5

Updated the version number to 1.0.5 and added author

JuanJavier1979 commented on public/templates/loop.php:
```diff
@@ -1,7 +1,7 @@
 <li class="isotope-item <?php echo esc_attr( $term_class ); ?> all">
-<?php if( has_post_thumbnail( $isotope_loop->ID ) ) : ?>
+<?php if( has_post_thumbnail( $this->isotope_loop->ID ) ) : ?>
```
Can you empty all params for has_post_thumbnail function?
As it runs inside a loop it does not need any post ID.

JuanJavier1979 commented on public/templates/loop.php:
```diff
@@ -1,7 +1,7 @@
+<?php if( has_post_thumbnail( $this->isotope_loop->ID ) ) : ?>
     <a href="<?php the_permalink(); ?>" title="<?php the_title(); ?>">
-        <?php the_post_thumbnail( $isotope_loop->ID, 'medium' ); ?>
+        <?php the_post_thumbnail( $this->isotope_loop->ID, 'medium' ); ?>
```
Can you empty all params for the_post_thumbnail function?
Same as previous comment.

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -24,7 +24,112 @@
+     * @since 1.0.5
+     * @var array
+     */
+    private $taxes = array();
```
Can you avoid the taxes word? You could better use taxonomies as it's the correct term.
Taxes might be confusing. :-)

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -43,53 +148,22 @@ public function __construct() {
 	 * @since    1.0.0
 	 *
+     * @since    1.0.5
+     * Moved shortcode arguement extraction logic in a function
```
Fix typo

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -24,7 +24,112 @@
+     * @since 1.0.5
+     * @var string
+     */
+    private $tax = '';
```
As previous comment.
Should we use taxonomy as variable name?

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -167,20 +210,26 @@ public function shortcode_isotope( $atts ) {
+            </ul>
+
+            <?php
+            if('yes'==$this->pagination){
```
Can you fix brace style
https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#brace-style

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+        $this->pagination=strtolower($pagination);
+        $this->taxes=explode(',',$this->tax);///this allows to have multiple comma separated taxonomies in shortcode
+
+        if( null == $this->id ) {
```
Missing space between the `if` and the opening parenthesis `(`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+            $this->id = 'wanna' . md5( date( 'jnYgis' ) );
+        }
+
+        if ('yes'==$this->pagination){
```
Missing spaces after `(` and before `)`
Missing spaces before and after `==`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+
+        if ('yes'==$this->pagination){
+            $this->paged = ( get_query_var( 'paged' ) ) ? get_query_var( 'paged' ) : 1; //get current page number
+        }else{
```
Missing spaces before and after `else`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+                )
+            );
+        } else {
+            $tax_query_array=array();
```
Can you add spaces before and after every `=`?

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+
+            $temp_tax_query_array=array();
+
+            $this->terms=explode(',',$this->term);///this allows to have multiple comma separated terms in shortcode
```
Fix space for `=`
Add spaces inside parenthesis.

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+
+            $this->terms=explode(',',$this->term);///this allows to have multiple comma separated terms in shortcode
+
+            foreach ($this->taxes as $tax){
```
Add spaces inside parenthesis.

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+            );
+        } else {
+            $tax_query_array=array();
+            $tax_query_array['relation']='OR';
```
Fix space for `=`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+            $tax_query_array=array();
+            $tax_query_array['relation']='OR';
+
+            $temp_tax_query_array=array();
```
Fix space for `=`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+            $this->terms=explode(',',$this->term);///this allows to have multiple comma separated terms in shortcode
+
+            foreach ($this->taxes as $tax){
+                $temp_tax_query_array['taxonomy']=$tax;
```
Fix space for `=`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+
+            foreach ($this->taxes as $tax){
+                $temp_tax_query_array['taxonomy']=$tax;
+                $temp_tax_query_array['field']='slug';
```
Fix space for `=`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+            foreach ($this->taxes as $tax){
+                $temp_tax_query_array['taxonomy']=$tax;
+                $temp_tax_query_array['field']='slug';
+                $temp_tax_query_array['terms']=$this->terms;
```
Fix space for `=`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+                $temp_tax_query_array['taxonomy']=$tax;
+                $temp_tax_query_array['field']='slug';
+                $temp_tax_query_array['terms']=$this->terms;
+                $tax_query_array[]=$temp_tax_query_array;
```
Fix space for `=`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+     * @param    array $atts Shortcode attributes
+     * @return   array $query_args query arguments to be used in WP Query
+     */
+    private function extract_isotope($atts){
```
Missing spaces after `(` and before `)`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -43,53 +148,22 @@ public function __construct() {
-            'tax'       => '',
-            'term'      => '',
-        ), $atts) );
+        $query_args=$this->extract_isotope($atts);
```
Fix spaces for `=`
Missing spaces after `(` and before `)`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -98,54 +172,23 @@ public function shortcode_isotope( $atts ) {
                     $term_class = '';
-                    foreach( (array)$tax_terms as $term ) {
-                        $term_class .= $term->slug . ' '; 
+                    foreach ($this->taxes as $tax){//handling of multiple taxonomies
```
Missing spaces after `(` and before `)`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+     * @param    int $pages amount of pages returned by WP_Query
+     * @param    int $range range of pages on pagination strip
+     */
+    private function pagination_isotope($pages = '', $range = 4){
```
Missing spaces after `(` and before `)`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+     * @param    int $range range of pages on pagination strip
+     */
+    private function pagination_isotope($pages = '', $range = 4){
+        $showitems = ($range * 2)+1;
```
Missing spaces after `(` and before `)`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+        $showitems = ($range * 2)+1;
+
+        global $paged;
+        if(empty($paged)) $paged = 1;
```
Missing spaces after `(` and before `)`

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+        global $paged;
+        if(empty($paged)) $paged = 1;
+
+        if($pages == '')
```
Missing spaces after `(` and before `)`
Use the `{` sign in the same line as the `if` statement. Please check previous brace styles comment.

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+        {
+            global $wp_query;
+            $pages = $wp_query->max_num_pages;
+            if(!$pages)
```
Missing spaces after `(` and before `)`
Please check brace styles

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+            }
+        }
+
+        if(1 != $pages)
```
Missing spaces after `(` and before `)`
Please check brace styles

JuanJavier1979 commented on public/wanna-isotope-shortcode.php:
```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+
+        if(1 != $pages)
+        {
+            echo "<div class=\"pagination prefix\"><span>Page ".$paged." of ".$pages."</span>";
```
Use translation functions to output copy. Like `esc_html_e` function.
For this line I would use the `printf` function. See examples here https://wordpress.stackexchange.com/questions/260512/how-to-print-translation-supported-text-with-html-url

ankgne committed:
Correcting for PR comments

Removed post ID from has_post_thumbnail function as it runs inside a
loop it does not need any post ID

ankgne committed:
Correcting for PR comments

1) Change taxes to taxonomies
2) Formatted if, else, elseif, foreach , for , while loops as per
comments/best practices link
3) Formated function calls , function implementation as per
comments/best practices link
4) Added internalization for pagination logic
5) Corrected a typo

ankgne commented:
Hi @JuanJavier1979 I have incorporated PR review comments and pushed the commit , with review comments incorporated . Kindly review it. 

Appreciate your comments on my previous push.
